src/axom/primal/tests/primal_bezier_patch.cpp

```diff
@@ -596,6 +596,224 @@ TEST(primal_bezierpatch, normal)
+    EXPECT_NEAR(partial_vv_exp[i], partial_vv_test[i], 1e-6);
+  }
+
+  auto partial_uv_test = bPatch.dudv(u, v);
+  VectorType partial_uv_exp = {-3.43768298544, 1.94078069698, 8.48995274462};
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(partial_uv_exp[i], partial_uv_test[i], 1e-6);
+    EXPECT_NEAR(partial_uv_exp[i], DuDv[i], 1e-6);
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_bezierpatch, second_derivative_linear)
+{
+  SLIC_INFO("Testing bezier Patch derivative calculation for linear");
+
+  const int DIM = 3;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using VectorType = primal::Vector<CoordType, DIM>;
+  using BezierPatchType = primal::BezierPatch<CoordType, DIM>;
+
+  const int order_u = 1;
+  const int order_v = 4;
+
+  // Test on a linear Bezier patch to make sure the correct values are zero
+  // clang-format off
+  PointType controlPoints[(order_u + 1) * (order_v + 1)] = {
+                  PointType {0, 0, 0}, PointType{0, 4,  0}, PointType{0, 8, -3}, PointType{0, 12, 1}, PointType{0, 16, 3},
+                  PointType {2, 0, 6}, PointType{2, 4,  5}, PointType{2, 8,  0}, PointType{4, 12, 2}, PointType{2, 16, 2}};
+  
+  double weights[(order_u + 1) * (order_v + 1)] = {
+                 1.0, 1.0, 1.0, 1.0, 1.0,
+                 1.0, 1.0, 1.0, 1.0, 1.0};
+  // clang-format on
+
+  BezierPatchType bPatch(controlPoints, weights, order_u, order_v);
+  const double u = 0.5, v = 0.5;
+
+  PointType eval;
+  VectorType Du, Dv, DuDu, DvDv, DuDv;
+  bPatch.evaluate_second_derivatives(u, v, eval, Du, Dv, DuDu, DvDv, DuDv);
+
+  auto evaluate_test = bPatch.evaluate(u, v);
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(eval[i], evaluate_test[i], 1e-6);
+  }
+
+  auto partial_u_test = bPatch.du(u, v);
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(Du[i], partial_u_test[i], 1e-6);
+  }
+
+  auto partial_v_test = bPatch.dv(u, v);
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(Dv[i], partial_v_test[i], 1e-6);
+  }
+
+  auto partial_uu_test = bPatch.dudu(u, v);
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(DuDu[i], partial_uu_test[i], 1e-6);
+  }
+
+  auto partial_vv_test = bPatch.dvdv(u, v);
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(DvDv[i], partial_vv_test[i], 1e-6);
+  }
+
+  auto partial_uv_test = bPatch.dudv(u, v);
+  for(int i = 0; i < 3; ++i)
+  {
+    EXPECT_NEAR(DuDv[i], DuDv[i], 1e-6);
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_bezierpatch, second_derivative_point)
+{
+  SLIC_INFO("Testing bezier Patch derivative calculation for point");
+
+  const int DIM = 3;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using VectorType = primal::Vector<CoordType, DIM>;
+  using BezierPatchType = primal::BezierPatch<CoordType, DIM>;
+
+  const int order_u = 0;
+  const int order_v = 4;
```

Comment:
Minor: would it also be worth testing `order_u=4` and `order_v=0` (and 1) against the same data?

Reply:
Yep! This is one of the tests I plan to put in soon. It's definitely necessary, since a lot of the tensor product de Casteljau work is done along the longer dimension first, and so there's unexplored code right now. It should be pretty easy to verify it works the way we want it to using `.swapAxes()`, which will save some lines of source code over near-identical tests for each of several combinations of order.